# Assignment 3

The US Department of Agriculture publishes price estimates for fruits and vegetables [online](https://www.ers.usda.gov/data-products/fruit-and-vegetable-prices/fruit-and-vegetable-prices/). The most recent estimates are based on a 2013 survey of US retail stores.

The estimates are provided as a collection of MS Excel files, with one file per fruit or vegetable. The `assignment3_data.zip` file contains the fruit and vegetable files in the directories `fruit` and `vegetables`, respectively.

__Exercise 1.1.__ Use pandas to extract the "Fresh" row(s) from the <strong style="color:#B0B">fruit</strong> Excel files. Combine the data into a single data frame. Your data frame should look something like this:

type       | food       | form   | price_per_lb | yield | lb_per_cup | price_per_cup
---------- | ---------- | ------ | ------------ | ----- | ---------- | -------------
fruit      | watermelon | Fresh1 | 0.333412     | 0.52  | 0.330693   | 0.212033
fruit      | cantaloupe | Fresh1 | 0.535874     | 0.51  | 0.374786   | 0.3938
vegetables | onions     | Fresh1 | 1.03811      | 0.9   | 0.35274    | 0.406868
...        |            |        |              |       |            |


It's okay if the rows and columns of your data frame are in a different order. These modules are especially relevant:

* [`str` methods](https://docs.python.org/2/library/stdtypes.html#string-methods)
* [`os`](https://docs.python.org/2/library/os.html)
* [`os.path`](https://docs.python.org/2/library/os.path.html)
* [pandas](http://pandas.pydata.org/pandas-docs/stable/): `read_excel()`, `concat()`, `.fillna()`, `.str`, plotting methods

Ask questions and search the documentation/web to find the functions you need.


In [14]:
import numpy as np
import pandas as pd
import os
#oldlist = os.listdir("assignment3_data/fruit/")
path = 'assignment3_data/fruit/'
lil = os.listdir(path)
#lil = [x for x in oldlist if "~" not in x] # list the files
#x = [pd.read_excel(path+y) for y in files]
#print lil

In [16]:
notwebscraped = [pd.read_excel(path+x) for x in lil]
for i in range(len(lil)):
    notwebscraped[i].iloc[:,5] = lil[i] # read in data

    


In [17]:
for i in range(len(lil)): #get the fruits
    notwebscraped[i].iloc[:,2] = "fruit"



In [18]:
for i in range(len(lil)):
    notwebscraped[i].iloc[:,5] = [x.rstrip(".xslsx") for x in notwebscraped[i].iloc[:,5]] #Loop to get fruit names

In [20]:
for i in range(len(lil)): #limit each df to 7 columns
    notwebscraped[i] = notwebscraped[i].iloc[:,:7]
notwebscraped

[   Apples—Average retail price per pound or pint and per cup equivalent, 2013  \
 0                                                Form                           
 1                                                 NaN                           
 2                                              Fresh1                           
 3                                         Applesauce2                           
 4                                               Juice                           
 5                                     Ready to drink3                           
 6                                             Frozen4                           
 7   1The USDA National Nutrient Database for Stand...                           
 8                                                 NaN                           
 9   2Excludes applesauce packed in individual cont...                           
 10                                                NaN                           
 11  3Includes r

In [21]:
# notwebscraped[1].iloc[:,0] = notwebscraped[i].iloc[:,0].dropna()
newdf = []
for i in range(len(lil)):
    notwebscraped[i].columns = ['form','price_per_lb','type','yield','lb_per_cup','food','price_per_cup'] #make them columns


In [28]:
newdf = [] #new data frame 
for i in range(len(lil)):
    newdf.append(notwebscraped[i])
ourdataframe = pd.concat(newdf)
outdataframe = ourdataframe.dropna()

In [25]:
datanon.iloc[:,0] = datanon.iloc[:,0].str.lower()  #lower the names



/Users/noa/anaconda/lib/python2.7/site-packages/pandas/core/indexing.py:549: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self.obj[item_labels[indexer[info_axis]]] = value


In [27]:
fruit = datanon[datanon['form'].str.contains("fresh")] #final fruit data
print fruit

      form price_per_lb   type yield lb_per_cup         food price_per_cup
2   fresh1      1.56752  fruit   0.9   0.242508        apple      0.422373
2   fresh1      3.04007  fruit  0.93   0.363763      apricot        1.1891
2   fresh1     0.566983  fruit  0.64   0.330693       banana      0.292965
2  fresh1       5.77471  fruit  0.96    0.31967  blackberrie       1.92292
2   fresh1      4.73462  fruit  0.95    0.31967   blueberrie       1.59318
2   fresh1     0.535874  fruit  0.51   0.374786   cantaloupe        0.3938
2   fresh1      3.59299  fruit  0.92   0.341717      cherrie       1.33455
2   fresh1     0.897802  fruit  0.49   0.462971   grapefruit      0.848278
2   fresh1      2.09383  fruit  0.96   0.330693        grape      0.721266
2   fresh1     0.796656  fruit  0.46   0.374786     honeydew      0.649077
2   fresh1      2.04468  fruit  0.76   0.385809         kiwi       1.03797
2   fresh1      1.37756  fruit  0.71   0.363763       mangoe      0.705783
2   fresh1      1.76115  

__Exercise 1.2.__ Reuse your code from exercise 1.1 to extract the "Fresh" row(s) from the <strong style="color:#B0B">vegetable</strong> Excel files.

Does your code produce the correct prices for tomatoes? If not, why not? Do any other files have the same problem as the tomatoes file?

You don't need to extract the prices for these problem files. However, make sure the prices are extracted for files like asparagus that don't have this problem.

In [273]:
veglist = os.listdir("/Users/ryanbrady/Desktop/141hw/vegetables")
veggies = [x for x in veglist if "~" not in x] # list the files


In [288]:
vegwebscraped = [pd.read_excel(x) for x in veggies]

for i in range(len(veglist)):
    vegwebscraped[i].iloc[:,5] = veggies[i] # read in data

In [289]:
for i in range(len(veglist)): #get the veg
    vegwebscraped[i].iloc[:,2] = "vegetable"

In [291]:
for i in range(len(veglist)):
    vegwebscraped[i].iloc[:,5] = [x.rstrip(".xslsx") for x in vegwebscraped[i].iloc[:,5]] #Loop to getveg names

In [294]:
for i in range(len(veglist)): #limit each df to 7 columns
    vegwebscraped[i] = vegwebscraped[i].iloc[:,:7]



In [295]:
for i in range(len(veglist)):
    vegwebscraped[i].iloc[:,0] = vegwebscraped[i].iloc[:,0].dropna()

In [305]:
vegdf = []
for i in range(len(veglist)):
    vegwebscraped[i].columns = ['form','price_per_lb','type','yield','lb_per_cup','food','price_per_cup'] #make them columns



In [306]:
for i in range(len(veglist)):
    vegdf.append(vegwebscraped[i])
vegdataframe = pd.concat(vegdf)

In [325]:
vegnona = vegdataframe[~vegdataframe['form'].isnull()]

In [327]:
vegnona.iloc[:,0] = vegnona.iloc[:,0].str.lower()  #lower the names



In [331]:
veggies =vegnona[vegnona['form'].str.contains("fresh")] #final veg data
new = 


veggies

My code does not produce the correct prices for tomatoes. Because tomatoes is not listed with fresh, but instead different descriptions of the tomatoes, the tomatoes get weeded out in the subsetting of the variables. Because many of the other vegetables are defined with multiple 

__Exercise 1.3.__ Remove rows without a price from the vegetable data frame and then combine the fruit and vegetable data frames. Make sure all columns of numbers are numeric (not strings).

__Exercise 1.4.__ Discuss the questions below (a paragraph each is sufficient). Use plots to support your ideas.

* What kinds of fruits are the most expensive (per pound)? What kinds are the least expensive?
* How do the price distributions compare for fruit and vegetables?
* Which foods are the best value for the price?
* What's something surprising about this data set?
* Which foods do you expect to provide the best combination of price, yield, and nutrition? A future assignment may combine this data set with another so you can check your hypothesis.